### PR TITLE
[KBFS-3561] Fix downloads on Android

### PR DESCRIPTION
**Pull request by jzila**
This might warrant a hotfix.

### CODE REVIEW

**Comment by taruti** [comment]
LGTM, but cannot test at the moment.

**Comment by songgao** [comment]
Would `mime.mimeType` not work?

**Comment by jzila** [comment]
Yes it works, I just like leaving explicit evidence that this is an immutable object in the code.

**Comment by songgao** [comment]
But it's not type-checked anymore right? For example, you could do `mime.get('MimeType')` and it'd pass flow, I think.

**Comment by jzila** [comment]
It appears to me that the type checking here is completely broken anyway, since Flow didn't catch this error in the first place.

**Comment by jzila** [comment]
Does type checking work for Immutable records e.g. with `.mimeType`?

**Comment by chrisnojima** [comment]
i agree, avoid get unless its a dynamic sub thing

**Comment by songgao** [comment]
> It appears to me that the type checking here is completely broken anyway, since Flow didn't catch this error in the first place.

That can't be an argument for giving up type checking there right? I think the reason why flow didn't cache it was probably because of the less typed generator function. But perhaps something like this would work:

```
    const mimeType: Types.MimeType = yield Saga.call(_loadMimeType, path)
```

Also I'm guessing the type checking must be working in `viewTypeFromMimeType`.

File: shared/fs/index.stories.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ const provider = Sb.createPropProviderWithCommon({
   }),
   ViewContainer: () => ({
     url: '/keybase/private/foo/bar.jpg',
-    mimeType: 'jpg',
+    mimeType: Constants.makeMime({mimeType: 'jpg'}),
```
Comment (songgao) [inline shared/fs/index.stories.js]: Did you set this to `jpg` instead of `image/jpeg` because otherwise it'd choose the image viewer?

**Comment by chrisnojima** [comment]
they flow type the same. .get isn't required for records and we shouldn't use it

File: shared/fs/index.stories.js (hunk 1 above)
Comment (jzila) [reply shared/fs/index.stories.js]: I set it with minimal changes to not be broken any more. Should it be `image/jpeg`?

**Pull request merged**